components/AlertsDashboard.vue

```diff
@@ -131,224 +131,230 @@ const emit = defineEmits(["reset-legend-visibility"]);
-          ),
-        },
-      });
-    }
+      if (type === "LineString") {
+        if (!map.value.getLayer(layerId)) {
+          map.value.addLayer({
+            id: layerId,
+            type: "line",
+            source: layerId,
+            filter: ["==", "$type", "LineString"],
+            paint: {
+              "line-color": [
+                "case",
+                ["boolean", ["feature-state", "selected"], false],
+                "#FFFF00",
+                strokeColor,
+              ],
+              "line-width": [
+                "case",
+                ["boolean", ["feature-state", "selected"], false],
+                5,
+                3,
+              ],
+              "line-opacity": 0.8,
+            },
+          });
+        }
+      }
+      resolve();
+    });
+  };
 
-    // Add a layer for previous alerts Polygons
-    if (!map.value.getLayer("previous-alerts-polygon")) {
-      map.value.addLayer({
-        id: "previous-alerts-polygon",
-        type: "fill",
-        source: "previous-alerts-polygon",
-        paint: {
-          "fill-color": [
-            "case",
-            ["boolean", ["feature-state", "selected"], false],
-            "#FFFF00",
-            "#FD8D3C",
-          ],
-          "fill-opacity": 0.5,
-        },
-      });
-    }
+  /**
+   * Loads and adds an image to the style. Returns a promise that resolves
+   * when the image is successfully loaded and added to the map.
+   */
+  const loadMapImage = (iconName: string, iconUrl: string): Promise<void> => {
+    return new Promise((resolve, reject) => {
+      map.value.loadImage(iconUrl, (error: Error, image: HTMLImageElement) => {
+        if (error) {
+          reject(error);
+          return;
+        }
 
-    // Add a stroke for previous alerts Polygons
-    if (!map.value.getLayer("previous-alerts-polygon-stroke")) {
-      map.value.addLayer({
-        id: "previous-alerts-polygon-stroke",
-        type: "line",
-        source: "previous-alerts-polygon",
-        paint: {
-          "line-color": [
-            "case",
-            ["boolean", ["feature-state", "selected"], false],
-            "#FFFF00",
-            "#FD8D3C",
-          ],
-          "line-width": 2,
-        },
+        if (!map.value.hasImage(iconName)) {
+          map.value.addImage(iconName, image);
+        }
+        resolve();
       });
-    }
-  }
+    });
+  };
 
-  // Check if the data contains LineString features for previous alerts
-  if (
-    geoJsonSource.previousAlerts.features.some(
-      (feature) => feature.geometry.type === "LineString",
-    )
-  ) {
-    // Add the previous alerts source to the map as LineStrings
-    if (!map.value.getSource("previous-alerts-linestring")) {
-      map.value.addSource("previous-alerts-linestring", {
+  /**
+   * Adds a GeoJSON point layer to the map using the geographicCentroid property,
+   * with specified icon.
+   */
+  const addAlertPointsLayer = async (
```

Comment:
I'm missing (due to my inexperience with vue) what exactly invokes `addAlertPointsLayer`, but ultimately what I'm wondering is:
Is there any chance of a race condition where two or more invocations to `addAlertsPointsLayer` are running at the same time?  For example it could be called before the previous call has finished. In this case some updates to ref state `map.value` could be lost.

Same question about `loadMapImage` making modifications to map.value (`.addImage()`) but I assume the answer is the same for both these.

Reply:
`addAlertsData` (which defines and calls these functions you are asking about) is called by `prepareMapCanvasContent`, which itself is called on a few occasions:

1. On mount (in `onMounted`).
2. If the basemap is changed (in `handleBaseMapChange`), once the map state is idle.

It's possible that somebody would try to change the basemap before the alerts data loads, but doing so rebuilds the entire `map` object anyway.